packages/signers/src/signers/utils/client.ts

```diff
@@ -135,10 +177,122 @@ export class AAClient extends SigningCosmWasmClient {
         funds: [],
       }),
     };
-    const tx = await this.sign(sender, [addMsg], fee, memo);
+
+    const {
+      gasPrice: gasPriceString,
+      gasAdjustment,
+      gasAdjustmentMargin,
+    } = xionGasValues;
+
+    const simmedGas = await this.simulate(sender, [addMsg], memo);
+    const gasPrice = GasPrice.fromString(gasPriceString);
+    const calculatedFee: StdFee = calculateFee(
+      simmedGas * gasAdjustment,
+      gasPrice,
+    );
+
+    let defaultFee: StdFee;
+    let gas = (
+      parseInt(calculatedFee.gas) * gasAdjustment +
+      gasAdjustmentMargin
+    ).toString();
+
+    const chainId = await this.getChainId();
+
+    if (/testnet/.test(chainId)) {
+      defaultFee = { amount: [{ amount: "0", denom: "uxion" }], gas: gas };
+    } else {
+      defaultFee = { amount: calculatedFee.amount, gas: gas };
+    }
+
+    const tx = await this.sign(sender, [addMsg], fee || defaultFee, memo);
```

Comment:
If this is repeated should we just make this a function?

Reply:
Good point. Fixed